commons-db/src/main/java/com/palantir/nexus/db/sql/SQLString.java

```diff
@@ -281,13 +282,13 @@ private static String canonicalizeString(String original, boolean removeAllWhite
             }
         }
 
-        if (cleanedString.charAt(cleanedIdx - 1) == ' ') {
+        while (cleanedIdx > 0 && Character.isWhitespace(cleanedString.charAt(cleanedIdx - 1))) {
```

Comment:
This one only needs to be an `if` since all extra whitespace will be compressed down to one above